src/render_engine/archive.py

```diff
@@ -36,7 +36,8 @@ def __init__(
         self.pages = pages
         self.template = template
         self.routes = routes
+        self.archive_index = archive_index
         self.title = title
 
-        if num_archive_pages > 1:
+        if archive_index:
```

Comment:
Supports the [new archive](https://github.com/kjaymiller/render_engine/discussions/186)